test/rules/promise-async/test.ts.lint

```diff
@@ -0,0 +1,38 @@
+const nonAsyncPromiseFunction = function(p: Promise<void>) { return p; };
```

Comment:
you need `declare class Promise<T>{}` for this to work. Otherwise, the type checker will not know what a Promise<> is and consider it to be `any`